### ISSUE TITLE
Question Regarding WebExtensions.Search.Search()

### ISSUE THREAD

**Issue by dragnilar** [mingyaulee/WebExtensions.Net#6]
Below is an example of how I am using the WebExtensions.Search.Search(SearchProperties searchProperties) method:

```
            try
            {
                Console.WriteLine($"Performing a search with query : {SearchQuery}");
                var searchProps = new SearchProperties {Query = SearchQuery};
                await WebExtensions.Search.Search(searchProps);
            }
            catch (Exception e)
            {
                Console.WriteLine("We had an error doing the search query.");
                Console.WriteLine(e);
            }
```

This always results in an exception:

```
Microsoft.JSInterop.JSException: Unable to find function search.search on object 'browser'.
blazor.webassembly.js:1 Error: Unable to find function search.search on object 'browser'.
```

My Manifest does have search declared.

If I work around it using JS Interop and call _chrome.search.query(...)_, the browser executes a search with my default search engine as expected. 

Am I doing something incorrectly or is this method not fully implemented?

**Comment by mingyaulee** [comment]
This is something that is not implemented yet. The reason being this repository is generated from Mozilla's definition of the extensions API, which according to their documentation, the [browser.search.search API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/search#browser_compatibility) is only supported in Firefox whereas in Chrome is it `chrome.search.query`.
If you are interested to fix this with a PR I will be happy to guide you, otherwise I will only be able to do it probably by next week.

**Comment by mingyaulee** [comment]
Awesome, what you need is exactly in the contributing guidelines under the section "Supporting missing/browser specific APIs".

**Comment by dragnilar** [comment]
@mingyaulee This should be taken care of now. I didn't get a chance to test it against the mock api you have in there until last night, which did work.

 I'm not sure if you want me to add an integration test for the search API but my instinct tells me there's a reason you didn't have one in there in the first place.

**Comment by mingyaulee** [comment]
Great! If you can add it I would appreciate it very much. A lot of tests are still missing now, some because I don't fully understand what each of the API does so I only added whatever that I could.

**Comment by dragnilar** [comment]
@mingyaulee the PR is up  https://github.com/mingyaulee/WebExtensions.Net/pull/7

**Comment by mingyaulee** [comment]
Released in v0.8.2
